src/librustc/ty/layout.rs

```diff
@@ -1076,6 +1080,12 @@ impl<'a, 'gcx, 'tcx> Layout {
                                         value: value
                                     });
                                 }
+                                CEnum { discr: value, .. } => {
+                                    return success(RawNullablePointer {
+                                        nndiscr: discr as u64,
+                                        value: Int(value)
```

Comment:
Maybe make this `match` produce the `Primitive` type and do the return after the `match`?